src/components/myProfile.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { createActionForCancel } from '../rocketsRedux/fetchData';
+
+const MyProfile = () => {
+  const dataRockets = useSelector((state) => state.rockets);
+  const dataMissions = useSelector((state) => state.missions);
+  const dispatch = useDispatch();
+  const cancelHandler = (id) => {
+    dispatch(createActionForCancel(id));
+  };
+  return (
+    <div className="profiles">
+      <div className="myMissions">
+        <div>
+          <h1>My Missions</h1>
+          {dataMissions.map((i) => i.reserved && (
+            <div className="joined">
+              <h2>{i.mission_name}</h2>
+              <Link to="/missions" className="leavetext"><button type="button" className="leave">To leave</button></Link>
+            </div>
+          ))}
+        </div>
+      </div>
+
+      <div className="myProfile">
+        <div className="rockets-container">
+          <h1>My Rockets</h1>
+          {dataRockets.map((i) => i.reserved && (
+            <div className="profile-rocket-div">
+              <h1>{i.rocket_name}</h1>
+              <button type="button" onClick={() => cancelHandler(i.id)}>Cancel Reservation</button>
+            </div>
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+};
+export default MyProfile;
```

Comment:
- [ ] When you click on the `Reverse Rocket`  or `Join Mission` button the data should be sent to the profile page in the MY Rockets field however when you visit the profile page and go back to my rocket page too all the action or data get lost, this applies to the Mission page too.

- You have to implement the state that will keep the data/update the state so that the data couldn't be lost when you reload the page or you navigate to the different pages as it is happening.

https://user-images.githubusercontent.com/55635977/168439763-a3edac79-e53d-4941-9711-52bffba72ff0.mp4